### DISCUSSION TITLE
Community Detection

### DISCUSSION THREAD

**Discussion by Cho-Geonwoo** [pyg-team/pytorch_geometric#3308] [Ideas]
# Feature Request

Features For [Community Detection](https://towardsdatascience.com/community-detection-algorithms-9bd8951e7dae) Task. Here are the features which are needed for implementing a community detection algorithm using GNN in my thought.

* Additional evaluation metrics for community detection such as modularity
* Train time node clustering function
* Way to handle Community Embedding

# Motivation

This is a paper which I am motivated from. 
[Deep Learning for Community Detection: Progress, Challenges and Opportunities](https://arxiv.org/abs/2005.08225)

# ETC

I'm not expert, but I'm really interested in community detection in networks using GNNs. If it's okay, I want to help on a pr related to community detection.

**Comment by rusty1s** [comment]
Great feature request. Do you have a specific paper/method in mind we shall start to integrate? Please feel free to contribute :)

**Reply by Cho-Geonwoo** [reply]
[GEMSEC](https://github.com/benedekrozemberczki/GEMSEC) which shows a SOTA performance for various Facebook dataset would be a great reference.

**Reply by Cho-Geonwoo** [reply]
Thx!

**Reply by rusty1s** [reply]
This sounds great. I think you can implement `GEMSEC` in a similar fashion as `torch_geometric.nn.models.Node2Vec`. Please feel free to reach out in case you encounter any issues.

**Reply by rjurney** [reply]
Did this ever get built?

**Reply by rusty1s** [reply]
Nope :(

**Reply by AslantheAslan** [reply]
I think using the "harmonic function" from NetworkX results in the best, still. I am not sure if that will be a contribution but harmonic function can be adapted in PyG as a feature. @Cho-Geonwoo did you try referring to that function ever?
